### PR TITLE
Remove reimbursement from Moving Expense Document

### PR DESCRIPTION
**Pull request by amitch23**
## Description

Reimbursement was mistakenly used as an association to a moving expense document. This PR removes the reimbursement association from everywhere, including:
- DB - moving expense document gets 2 new fields with existing reimbursement data before the reimbursement row is deleted
- Handlers and tests
- Client forms, schema, etc.
- e2e tests

It also removes the unused update moving expense document in swagger.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:
- With an approved move and ppm, upload a document from the "Request payment" btn on the service member landing page. Ensure that the document is uploaded. On the office side, upload a document by clicking on the document icon on the basics page, edit an existing document and change the status to 'OK'. You should see the document in the document viewer. Go to the PPM expense summary panel - you should see the document expense data there.

## Code Review Verification Steps

* [ ] End to end tests pass (`make e2e_test`).
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/159538182) for this change

### CODE REVIEW

File: migrations/20180807232856_remove-reimbursement-from-expense-doc.up.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+UPDATE moving_expense_documents
+SET payment_method = 'OTHER'
+WHERE payment_method = 'MIL_PAY';
+
+-- Delete soon-to-be orphaned reimbursements
+DELETE FROM reimbursements WHERE id = (SELECT reimbursement_id from moving_expense_documents WHERE moving_expense_documents.reimbursement_id = reimbursements.id);
+
+
+-- Delete reimbursement_id column moving_expense_documents
+ALTER TABLE moving_expense_documents DROP COLUMN reimbursement_id CASCADE;
```
Comment (amitch23) [inline migrations/20180807232856_remove-reimbursement-from-expense-doc.up.sql]: This cascade doesn't actually delete the reimbursements, despite what the postgres docs may say. The line above will take care of deleting orphaned reiumbursements - is there a better way?

Comment (stangah) [reply migrations/20180807232856_remove-reimbursement-from-expense-doc.up.sql]: It looks like we never set up the foreign key constraint on this column when we created the table, so CASCADE doesn't think there's anything worth deleting. What you have here is good!

File: pkg/handlers/move_documents.go [hunk 2]
```diff
@@ -132,24 +142,25 @@ func (h UpdateMoveDocumentHandler) Handle(params movedocop.UpdateMoveDocumentPar
...
-		reimbursementAmt := unit.Cents(*payload.Reimbursement.RequestedAmount)
-		reimbursementMethod := models.MethodOfReceipt(*payload.Reimbursement.MethodOfReceipt)
+		requestedAmt := unit.Cents(payload.RequestedAmountCents)
+		paymentMethod := payload.PaymentMethod
 		if moveDoc.MovingExpenseDocument == nil {
 			// But we don't have one, so create it to be saved later
-			reimbursement := models.BuildRequestedReimbursement(
-				reimbursementAmt,
-				reimbursementMethod)
+			// reimbursement := models.BuildRequestedReimbursement(
```
Comment (stangah) [inline pkg/handlers/move_documents.go]: Can this go?

File: pkg/models/move.go [hunk 3]
```diff
@@ -311,19 +313,19 @@ func (m Move) CreateMovingExpenseDocument(
 		}
 
 		// Save the reimbursement to get an ID.
-		if verrs, err := db.ValidateAndCreate(&reimbursement); verrs.HasAny() || err != nil {
-			responseVErrors.Append(verrs)
-			responseError = errors.Wrap(err, "Error Creating Moving Expense Advance")
-			return transactionError
-		}
+		// if verrs, err := db.ValidateAndCreate(&reimbursement); verrs.HasAny() || err != nil {
```
Comment (stangah) [inline pkg/models/move.go]: Same here

File: swagger/internal.yaml [hunk 4]
```diff
@@ -464,7 +464,7 @@ definitions:
     properties:
       MIL_PAY:
         type: integer
-      OTHER_DD:
```
Comment (stangah) [inline swagger/internal.yaml]: Should this change too?

File: swagger/internal.yaml [hunk 5]
```diff
@@ -621,31 +634,6 @@ definitions:
       - upload_ids
       - title
       - move_document_type
-  UpdateGenericMoveDocumentPayload:
```
Comment (stangah) [inline swagger/internal.yaml]: I thought we were keeping these around to validate some client forms, are we not?

Comment (amitch23) [reply swagger/internal.yaml]: We aren't. We're using the `MoveDocumentPayload` and then passing in `required` props for each form swaggerField instead. e.g. https://github.com/transcom/mymove/blob/master/src/scenes/Office/DocumentViewer/DocumentDetailPanel.jsx#L107

File: pkg/handlers/move_documents.go (hunk 2 above)
Comment (amitch23) [reply pkg/handlers/move_documents.go]: done

File: pkg/models/move.go (hunk 3 above)
Comment (amitch23) [reply pkg/models/move.go]: done

File: migrations/20180807232856_remove-reimbursement-from-expense-doc.up.sql [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+-- Add data from existing reimbursement fields to moving_expense_documents
+UPDATE moving_expense_documents
+SET requested_amount_cents = reimbursements.requested_amount,
+  payment_method = reimbursements.method_of_receipt
+FROM reimbursements
+WHERE reimbursements.id = moving_expense_documents.reimbursement_id;
+
+-- Milpay is not allowed as a requested payment method, so set to 'OTHER'
+UPDATE moving_expense_documents
```
Comment (pjdufour-truss) [inline migrations/20180807232856_remove-reimbursement-from-expense-doc.up.sql]: I'm not sure this does what you intend.  Should you be using `or`?

Comment (amitch23) [reply migrations/20180807232856_remove-reimbursement-from-expense-doc.up.sql]: Yes I meant to use OR, good catch.

**Comment by pjdufour-truss** [comment]
I added a comment to migrations file

**Pull request merged**